### ISSUE TITLE
How to train LSTM or BiLSTM for sequence to label problem using arbitrary length multivariate time series data?

### ISSUE THREAD

**Issue by MNFuad** [keras-team/keras#11429]
I'm working on a sequence labeling task. The input data dimension  is 3xT, where 3 is the number of channels and T is the data length (e.g. time). I tried a two layer LSTM, it seems working fine for training single subject data. However, I have multi-subject problem (i.e. Nx3xT), where N is the number of subjects.

1. How could I train the network for all subjects at once? Do I have to concatenate the data? which is going to be a huge length and impossible to load to memory. Or I am missing something with this code.

2. For the batch_size, the model structure does accept only the batch_size which is dividable by the data shape (X_train.shape[0]) which is arbitrary length (e.g .1901 or 2019 , etc.). Same for test data, it has to be dividable by batch_size. Is there any way to make it work with no zero-padding.

`batch_input_shape = 1`
`X_train.shape[1] = 3`
`X_train.shape[2] = 1`
`dimof_output = 4`
`batch_size = 1`
`model = Sequential()`
`model.add(LSTM(units=25, batch_input_shape=[batch_size, X_train.shape[1], X_train.shape[2]], stateful=True, dropout=0.2, recurrent_dropout=0.2))`
`model.add(Dense(dimof_output, kernel_initializer='uniform', activation='softmax'))`
`model.compile(loss='categorical_crossentropy', optimizer='adam', metrics=['accuracy'])`

I replaced the LSTM layer with BiLSTM and shows a lower performance.
 
`model.add(Bidirectional(LSTM(units=35, stateful=True), batch_input_shape=[batch_size, X_train.shape[1], X_train.shape[2]]))`
